### PR TITLE
fix(ci/cd): fix release pipeline and adding tests

### PR DESCRIPTION
**Pull request by paologallinaharbur**
Following the PR https://github.com/newrelic/nri-postgresql/pull/135/commits/e7e437d6c4eac2d6fe50dae4d64a123477dc3060

### CODE REVIEW

File: .github/workflows/push_pr.yml [hunk 1]
```diff
@@ -97,16 +98,68 @@ jobs:
...
+      GPG_PASSPHRASE: ${{ secrets.OHAI_GPG_PASSPHRASE }}
+      GPG_PRIVATE_KEY_BASE64: ${{ secrets.OHAI_GPG_PRIVATE_KEY_BASE64 }} # base64 encoded
     steps:
       - uses: actions/checkout@v4
-      - name: Login to DockerHub
-        if: ${{env.DOCKER_LOGIN_AVAILABLE}}
-        uses: docker/login-action@v3
-        with:
-          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
-          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
```
Comment (kang-makes) [inline .github/workflows/push_pr.yml:110]: NIT: I remember you saying this helped to overcome the rate limit.

Does it still apply? Should we still remove these lines?

File: .github/workflows/push_pr.yml [hunk 2]
```diff
@@ -97,16 +98,68 @@ jobs:
...
+            echo NO_SIGN=true >> $GITHUB_ENV
+          else
+            printf "%s" "$PFX_CERTIFICATE_BASE64" | base64 -d - > wincert.pfx
+          fi
+      - name: Extract .exe
+        shell: pwsh
+        run: build\windows\extract_exe.ps1 "$env:INTEGRATION" ${{ matrix.goarch }} "$env:TAG"
+      - name: Create MSI
+        shell: pwsh
+        run: build\windows\package_msi.ps1 -integration "$env:INTEGRATION" -arch ${{ matrix.goarch }} -tag "$env:TAG" -pfx_passphrase "$env:PFX_PASSPHRASE" -pfx_certificate_description "$env:PFX_CERTIFICATE_DESCRIPTION"
```
Comment (kang-makes) [inline .github/workflows/push_pr.yml:165]: NIT: Missing EOF line.

**Review by kang-makes — APPROVED** [review]
Awesome!

File: .github/workflows/push_pr.yml (hunk 1 above)
Comment (paologallinaharbur) [reply .github/workflows/push_pr.yml:110]: you are right!

**Pull request merged**